### ISSUE TITLE
[I18n] Document process for translating strings

### ISSUE THREAD

**Issue by ianegordon** [material-components/material-components-ios#1259]
Our components have a variety of strings that must be translated into our supported languages.

We should document this process and include it in our contributing info.

**Comment by Arcank** [comment]
Kindly ping. Is there a first batch (even manual) that could be run? For example for https://github.com/material-components/material-components-ios/blob/develop/components/CollectionCells/src/MaterialCollectionCells.bundle/Resources/en.lproj/MaterialCollectionCells.strings

Thanks!

**Comment by ianegordon** [comment]
Bonjour Louis,

I'll be cutting the release this week and I'm planning on taking a pass at extracting our strings as part of the process.  The current thinking is to include the extracting of the strings as a step in our release guide.  I'll update here as I go through the release.

**Comment by Arcank** [comment]
Any new dev with yeasterday's release? Thanks!

**Comment by ianegordon** [comment]
tl;dr: I18n is broken.  We should fix it before we dig ourselves too deep.  I will pull together a design doc.


Current State:
Our existing Localization technique is generated from an internal non-public script.
Our existing Localization technique does not work with the Apple recommended genstrings or xcodebuild --exportLocalization.

Since we are an embedded framework, we need to include a table and bundle in our NSLocalizedString() calls.
The way we determine the bundle to pass in is non-standard.
The table value is likely not needed.

We need to be careful about how we declare and utilize our strings.
//GOOD
NSString *formatCharacterCountString = 
    NSLocalizedString(@"%d of %d", nil, bundle, @"Character count: current / maximum");
NSString *actualCharacterCountString = 
    [NSString stringWithFormat:formatCharacterCountString, textfield.text.count, MAX_LENGTH];
characterCountLabel.text = actualCharacterCountString;
	
//BAD
characterCountLabel.text = 
    [NSString stringWithFormat:@"%d of %d", textfield.text.count, NAME_MAX_LENGTH];


Our current script creates a MaterialComponent.h and MaterialComponent_string.h which are incompatible with Apple's tools.  For instance, they fail on the following (common) pattern:
NSString *key = kMaterialPageControlStringTable[kStr_MaterialPageControlAccessibilityLabel];
NSString *localizedString = 
  NSLocalizedStringFromTableInBundle(key, kMaterialPageControlStringsTableName, 
    [self bundle], @"page {number} of {total number}");
This issue is with key, "Argument is not a literal string."

Table name is discarded by genstrings.

**Comment by Arcank** [comment]
I have no idea how I closed that one :-|

Is there a design doc for the new process?

**Comment by Arcank** [comment]
Kindly ping, what is the translations status?

**Comment by ianegordon** [comment]
xcodebuild -importexportLocalizations has some limitations that genstrings does not.
@willlarche was able to discuss with Apple @ WWDC
Let's catch up offline on the best way forward.
